### ISSUE TITLE
DNF should give 0 points instead of position

### ISSUE THREAD

**Issue by KujawskiSzymon** [pjsavola/formulad#4]
When DNFing the car/player should get 0 points
Right not it is possible to dnf as the last man standing and get all the points
The code checks DNF with just hitpoints like:
dnfRaces.setText(Long.toString(filteredResults.stream().filter(r -> r.remainingHitpoints <= 0).count()));

maybe 
    private int getPoints(boolean isDnf, int position) {
        if (!isDnf) {
        return position > pointDistribution.length ? 0 : pointDistribution[position - 1];
        }
        else {
            return 0;
        }
    }
should check if player HP is  above 0 and then return 0 points if player DNF'ed

Ive tried to add DNF by myself but there are lines like
        final int total2 = positions.get(id2).stream().mapToInt(this::getPoints).sum();
and im not able to understand whats going on, I also dont have java experience

**Comment by pjsavola** [comment]
Yeah, the way to change this would likely be to change Season.createStandingsPanel() method so that you'd extract DNF status (or hitpoints) from FinalStandings similarly as positions are extracted. The best way to do this might be to create a new data pair which contains both list of positions and list of booleans for DNF statuses. And then you could modify getPoints-method as you suggested.

**Comment by KujawskiSzymon** [comment]
Ok, i did it, i can make a pull request after previous being checked and accepted (i guess patch set will be needed)

**Comment by pjsavola** [comment]
Thanks, I will check that PR during my Christmas vacation.
